app/controllers/providers/omniauth_callbacks_controller.rb

```diff
@@ -5,11 +5,17 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
     before_action :check_provider_is_enrolled, only: [:saml]
 
     def saml
-      provider = Provider.from_omniauth(auth_hash)
-
-      sign_in_and_redirect(
-        provider, event: :authentication
-      )
+      if Provider.active_offices?(auth_hash)
+        provider = Provider.from_omniauth(auth_hash)
+
+        sign_in_and_redirect(
+          provider, event: :authentication
+        )
+      elsif offices_in_list?
```

Comment:
this feels a bit odd but i think i understand it

```
if they have any active office (or NOT any inactive offices)
  sign them in 👍 
elsif auth hash has enrolled offices (or email enrolled or ALL enabled)
  redirect to "you need an active office code" <--- This feels odd
else
  redirect to "not enrolled yet" 👍 
end 
```

Given
- inactive offices BBBBBB and CCCCCC
- enrolled offices AAAAAA, BBBBBB and CCCCCC

if user A has AAAAAA, BBBBBB and CCCCC then they get signed in
if user B has  BBBBBB and CCCCCC then they fall through but because they are enrolled with inactive office codes they get redirect to the new page.
if user D has DDDDDD they fall through and get not enrolled page

When we switch to ALL for crm5 and unenroll AAAAAA then this should still work as expected user A 👍 and should still redirect B 👍 , but not D 👍 

I think this will work ok therefore but perhaps change the name of `offices_in_list?` to `[inactive_]office_enrolled?`

Reply:
yeah fair point, I did find that method name a bit awkward. There's probably a cleaner way to do all of this but I didn't want to redesign how it's currently done for no good reason. I've updated the method name